### PR TITLE
ILS reports changes

### PR DESCRIPTION
**Pull request by lwyszomi**
Need migration

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 15 errors:

* custom/ilsgateway/filters.py, line 107 - W292 no newline at end of file
* custom/ilsgateway/models.py, line 7 - F401 'OneToOneField' imported but unused
* custom/ilsgateway/tanzania/__init__.py, line 17 - F401 'latest_status' imported but unused
* custom/ilsgateway/tanzania/__init__.py, line 122 - E125 continuation line with same indent as next logical line
* custom/ilsgateway/tanzania/__init__.py, line 136 - E231 missing whitespace after ','
* custom/ilsgateway/tanzania/__init__.py, line 138 - E231 missing whitespace after ','
* custom/ilsgateway/tanzania/__init__.py, line 141 - E226 missing whitespace around arithmetic operator
* custom/ilsgateway/tanzania/__init__.py, line 142 - E226 missing whitespace around arithmetic operator
* custom/ilsgateway/tanzania/__init__.py, line 158 - E301 expected 1 blank line, found 0
* custom/ilsgateway/tanzania/reports/delivery.py, line 141 - E231 missing whitespace after ','
* custom/ilsgateway/tanzania/reports/mixins.py, line 161 - E501 line too long (134 > 115 characters)
* custom/ilsgateway/views.py, line 293 - E302 expected 2 blank lines, found 1
* custom/ilsgateway/views.py, line 299 - E302 expected 2 blank lines, found 1
* custom/ilsgateway/views.py, line 323 - W292 no newline at end of file
* custom/logistics/tasks.py, line 87 - E302 expected 2 blank lines, found 1

File: custom/ilsgateway/tanzania/__init__.py [hunk 1]
```diff
@@ -106,7 +115,73 @@ class ILSMixin(object):
     report_delivery_url = None
 
 
-class MultiReport(SqlTabularReport, ILSMixin, CustomProjectReport, ProjectReportParametersMixin, MonthYearMixin):
+class ILSDateSpan(DateSpan):
+
+    @classmethod
+    def from_month_or_quarter(cls, month_or_quarter=None, year=None, format=DEFAULT_DATE_FORMAT,
+        inclusive=True, timezone=pytz.utc):
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/__init__.py]: E125 continuation line with same indent as next logical line

File: custom/ilsgateway/tanzania/__init__.py [hunk 2]
```diff
@@ -106,7 +115,73 @@ class ILSMixin(object):
...
+            year = datetime.datetime.date.today().year
+        assert isinstance(month_or_quarter, int) and isinstance(year, int)
+        if month_or_quarter < 0:
+            quarters = list(rrule.rrule(rrule.MONTHLY,
+                                        bymonth=(1, 4, 7, 10),
+                                        bysetpos=-1,
+                                        dtstart=datetime(year, 1, 1),
+                                        count=5))
+
+            start = quarters[month_or_quarter*-1-1]
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/__init__.py]: E226 missing whitespace around arithmetic operator

File: custom/ilsgateway/tanzania/__init__.py [hunk 3]
```diff
@@ -106,7 +115,73 @@ class ILSMixin(object):
...
+            end = quarters[month_or_quarter*-1] - relativedelta(days=1)
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/__init__.py]: E226 missing whitespace around arithmetic operator

File: custom/ilsgateway/tanzania/__init__.py [hunk 4]
```diff
@@ -116,6 +191,13 @@ class MultiReport(SqlTabularReport, ILSMixin, CustomProjectReport, ProjectReport
     base_template = 'ilsgateway/base_template.html'
 
     @property
+    def location(self):
```
Comment (czue) [inline custom/ilsgateway/tanzania/__init__.py]: might want to `memoize` this if referenced in more than one place

**Comment by czue** [comment]
did a quick pass and looks good. i don't think the migration will be a problem since it's just making a new table.

**Pull request merged**